test/SILOptimizer/mandatory_inlining_and_combine.sil

```diff
@@ -0,0 +1,317 @@
+  %9 = function_ref @reabstractionThunk : $@convention(thin) <τ_0_0> (@in Builtin.Int32, @owned @callee_owned (Builtin.Int32) -> @out τ_0_0) -> @out τ_0_0
+  // INLINE-CHECK: partial_apply
+  // COMB-CHECK-NOT: partial_apply
+  %10 = partial_apply %9<U>(%2) : $@convention(thin) <τ_0_0> (@in Builtin.Int32, @owned @callee_owned (Builtin.Int32) -> @out τ_0_0) -> @out τ_0_0
+  dealloc_stack %6 : $*Builtin.Int32
+  strong_release %2 : $@callee_owned (Builtin.Int32) -> @out U
+  %14 = tuple ()
+  // COMB-CHECK: return
+  // INLINE-CHECK: return
+  return %14 : $()
+}
+
+// CHECK-LABEL: sil [transparent] @reabstractionThunk
+sil [transparent] @reabstractionThunk : $@convention(thin) <U> (@in Builtin.Int32, @owned @callee_owned (Builtin.Int32) -> @out U) -> @out U {
+bb0(%0 : $*U, %1 : $*Builtin.Int32, %2 : $@callee_owned (Builtin.Int32) -> @out U):
+  %3 = load %1 : $*Builtin.Int32
+  %4 = apply %2(%0, %3) : $@callee_owned (Builtin.Int32) -> @out U
+  // CHECK: return
+  return %4 : $()
+}
+
+// CHECK-LABEL: sil [transparent] @inner_transparent
+sil [transparent] @inner_transparent : $@convention(thin) (@owned @callee_owned () -> Builtin.Int8) -> Builtin.Int8 {
+// CHECK: bb0
+bb0(%0 : $@callee_owned () -> Builtin.Int8):
+  strong_retain %0 : $@callee_owned () -> Builtin.Int8
+  %3 = apply %0() : $@callee_owned () -> Builtin.Int8
+  strong_release %0 : $@callee_owned () -> Builtin.Int8
+  // CHECK: return
+  return %3 : $Builtin.Int8
+}
+
+// CHECK-LABEL: sil [transparent] @outer_transparent
+sil [transparent] @outer_transparent : $@convention(thin) (@owned @callee_owned () -> Builtin.Int8) -> Builtin.Int8 {
+// CHECK: bb0
+bb0(%0 : $@callee_owned () -> Builtin.Int8):
+  %2 = function_ref @inner_transparent : $@convention(thin) (@owned @callee_owned () -> Builtin.Int8) -> Builtin.Int8
+  strong_retain %0 : $@callee_owned () -> Builtin.Int8
+  %4 = apply %2(%0) : $@convention(thin) (@owned @callee_owned () -> Builtin.Int8) -> Builtin.Int8
+  strong_release %0 : $@callee_owned () -> Builtin.Int8
+  // CHECK: return
+  return %4 : $Builtin.Int8
+}
+
+// Mandatory combine should remove everything
+// INLINE-CHECK-LABEL: sil @testouter_transparent
+// CHECK-LABEL: sil @testouter_transparent
+sil @testouter_transparent : $@convention(thin) (Builtin.Int8) -> Builtin.Int8 {
+// CHECK: bb0
+// CHECK-NEXT: return
+// INLINE-CHECK: bb0
+// INLINE-CHECK: function_ref
+// INLINE-CHECK: partial_apply
+bb0(%0 : $Builtin.Int8):
+  %2 = function_ref @outer_transparent : $@convention(thin) (@owned @callee_owned () -> Builtin.Int8) -> Builtin.Int8
+  %3 = function_ref @identity_closure : $@convention(thin) (Builtin.Int8) -> Builtin.Int8
+  %4 = partial_apply %3(%0) : $@convention(thin) (Builtin.Int8) -> Builtin.Int8
+  %5 = apply %2(%4) : $@convention(thin) (@owned @callee_owned () -> Builtin.Int8) -> Builtin.Int8
+  return %5 : $Builtin.Int8
+}
+
+sil shared [transparent] @identity_closure : $@convention(thin) (Builtin.Int8) -> Builtin.Int8 {
```

Comment:
I'm ignoring the fact that this function no longer gets removed because it seems inline with the update in #28484